probe/process/reporter.go

```diff
@@ -21,20 +23,30 @@ const (
 
 // Reporter generates Reports containing the Process topology.
 type Reporter struct {
-	scope   string
-	walker  Walker
-	jiffies Jiffies
+	scope          string
+	walker         Walker
+	jiffies        Jiffies
+	maxMemory      MaxMemory
+	openFilesLimit OpenFilesLimit
 }
 
 // Jiffies is the type for the function used to fetch the elapsed jiffies.
 type Jiffies func() (uint64, float64, error)
 
+// MaxMemory is the type for the function used to fetch the hosts' max memory.
+type MaxMemory func() (uint64, error)
+
+// OpenFilesLimit is the type for the function used to fetch the hosts' open files limit.
```

Comment:
Ah, yes in fact. Have to get the information that from the procfs. Will change that.